File: solutions/zig/square-root/1/square_root.zig
pub fn squareRoot(radicand: usize) usize {
    var left: usize = 1;
    var right = radicand;
    while (left + 1 < right) {
        const middle = (left + right) / 2;
        if (middle * middle <= radicand) {
            left = middle;
        } else {
            right = middle;
        }
    }
    return left;
}

File: solutions/julia/square-root/2/square-root.jl
function square_root(n)
    left = 1
    right = n
    while left < right
        middle = (left + right + 1) ÷ 2
        sq = middle * middle
        if sq <= n && sq != 0
            left = middle
        else
            right = middle - 1
        end
    end
    left
end
